pkg/component/component.go

```diff
@@ -145,6 +144,29 @@ func (r *RuntimeSpecs) ToComponents(
 	return components, nil
 }
 
+// varsForPlatform sets the runtime variables that are available in the
+// input specification runtime checks. This function should always be
+// edited in sync with the documentation in specs/README.md.
+func varsForPlatform(platform PlatformDetail) (*transpiler.Vars, error) {
+	hasRoot, err := utils.HasRoot()
+	if err != nil {
+		return nil, err
+	}
+	return transpiler.NewVars("", map[string]interface{}{
+		"runtime": map[string]interface{}{
+			"platform": platform.String(),
+			"os":       platform.OS,
+			"arch":     platform.Arch,
+			"family":   platform.Family,
+			"major":    platform.Major,
+			"minor":    platform.Minor,
+		},
+		"user": map[string]interface{}{
+			"root": hasRoot,
```

Comment:
Should `uid` and `gid` be included in this map?

Reply:
Yes, I meant to leave a note there for review. They aren't currently used and I can't imagine any reasonable way to use them (they don't work at all on windows but even on mac and linux they're hardly stable -- a deployment-specific policy _might_ want conditions that use them, but not a spec file). I checked with Blake and he said removing them should be ok since nothing currently depends on them.